### PR TITLE
settings and clearing data

### LINKED ISSUE #222: cypress data/reset endpoint
- Add code to delete/reset data for cypress tests.
- This will be a guardrailed management command

### PR DESCRIPTION
**Pull request by raprasad**
This PR has two main changes:
1. The `settings.py `file is now a `settings/` directory. Within this directory:
    - `base.py` is the "core" settings file 
    - Other files such as `development.py` and `cypress_settings.py` read from base.py and update specific values
 2. (danger) For cypress tests, a `clear_test_data` managment command has been added which deletes nearly all data from the database.
    - This command is also available via API
    - The command is in a new app `opendp_apps.cypress_utils`
    - The command is only available when the new app is in INSTALLED_APPS
    - To isolate this command for use only in cypress tests, **several steps have been taken as described in the `cypress_settings.py` file**
      - **See comments in this file: [cypress_settings.py](https://github.com/opendp/dpcreator/blob/9e3faa05059fbf144d47b8068eddabaef189d192/server/opendp_project/settings/cypress_settings.py)**
    - In addition, there are unit tests to check that the command has limited availability

### CODE REVIEW

File: server/opendp_project/settings/cypress_settings.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+"""
+import os
+
+from opendp_project.settings.base import *
+
+DEBUG = True
+
+# (1) This app includes the "clear_test_data" management command
+INSTALLED_APPS += ['opendp_apps.cypress_utils']
```
Comment (ecowan) [inline server/opendp_project/settings/cypress_settings.py:22]: This is a really clever way to protect the system 👍

File: server/opendp_project/urls_cypress.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"""
+!DANGER!: These urlpatterns should only be used in test.
+Please read server/opendp_project/settings/cypress_settings for an explanation
+"""
+from django.urls import path, include
+
+from opendp_project.urls import urlpatterns
+from opendp_apps.cypress_utils.check_setup import are_cypress_settings_in_place
+
+if are_cypress_settings_in_place:
```
Comment (ecowan) [inline server/opendp_project/urls_cypress.py:10]: Is there a way for us to test spinning up an instance that shouldn't have this endpoint, then verifying that trying to hit it returns a 404, and similarly that the management command doesn't exist?

**Review by ecowan — APPROVED** [review]
Left one idea about testing, looks good otherwise.

File: server/opendp_project/urls_cypress.py (hunk 2 above)
Comment (raprasad) [reply server/opendp_project/urls_cypress.py:10]: There are several tests here that receive a 404 when hitting the endpoint and also test the managment command:
  - https://github.com/opendp/dpcreator/blob/9e3faa05059fbf144d47b8068eddabaef189d192/server/opendp_apps/cypress_utils/tests.py

Comment (ecowan) [reply server/opendp_project/urls_cypress.py:10]: Thanks for pointing that out, looks great

**Pull request merged**